src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java

```diff
@@ -14,37 +16,17 @@
 
 public class ConfigurableExtensionFilter extends AbstractDescribableImpl<ConfigurableExtensionFilter> {
 
-    @Extension
-    public static final ExtensionFilter EXTENSION_FILTER = new ExtensionFilter() {
-
-        @Override
-        public <T> boolean allows(Class<T> tClass, ExtensionComponent<T> tExtensionComponent) {
-            // Don't exclude myself
-            if (tExtensionComponent.getInstance() == this) return true;
-
-            return DESCRIPTOR.allows(tClass.getName())
-                    && DESCRIPTOR.allows(
-                            tExtensionComponent.getInstance().getClass().getName());
-        }
-    };
-
-    @Extension
-    public static final DescriptorVisibilityFilter DESCRIPTOR_FILTER = new DescriptorVisibilityFilter() {
-
-        @Override
-        public boolean filter(@CheckForNull Object context, @Nonnull Descriptor descriptor) {
-            Class contextClass = context == null ? null : context.getClass();
-            return DESCRIPTOR.allows(contextClass, descriptor.getClass().getName());
-        }
-    };
-
-    @Extension
-    public static final DescriptorImpl DESCRIPTOR = new DescriptorImpl();
+    public static /*almost final*/ DescriptorImpl DESCRIPTOR;
```

Comment:
`DESCRIPTOR should be package protected`. So, I would recommend to remove public